### DISCUSSION TITLE
End of life Qt5 - change to Qt6

### DISCUSSION THREAD

**Discussion by sjib** [teksi/Home#105] [Show and tell]
At the QGIS user meeting in Bern it was announced that QGIS is in the change process to switch to Qt6 instead of Qt5 (announced end of life 2025).

Qt6 can be additionally installed thru OS4GEO4W to an existing QGIS installation

- [ ] Test existing plugins
- [ ] Include these requirements in future development

Background: https://github.com/qgis/QGIS-Enhancement-Proposals/issues/292

**Comment by ponceta** [comment]
https://github.com/teksi/wastewater/issues/429

**Comment by sjib** [comment]
Documentation how to deal with Qt6 compatibility on future QGIS Releases:
https://github.com/opengisch/QgisModelBaker/issues/985#issue-2683221598

(Original source https://lists.osgeo.org/pipermail/qgis-developer/2024-January/066417.html)

**Comment by sjib** [comment]
https://github.com/qgis/QGIS/wiki/Plugin-migration-to-be-compatible-with-Qt5-and-Qt6/

**Comment by sjib** [comment]
https://www.qgis.org/project/visual-changelogs/visualchangelog336/#feature-python-support-in-qt6-builds

**Comment by sjib** [comment]
Idea for qt6 compatibility tests: https://github.com/opengisch/qfieldsync/pull/739#issuecomment-3734315105

**Comment by sjib** [comment]
@ponceta Support tools:

https://github.com/qgis/pyqgis4-checker
and even better
https://github.com/qgis/QGIS/wiki/Plugin-migration-to-be-compatible-with-Qt5-and-Qt6

**Reply by ponceta** [reply]
Nice! I have to check with latest Master to see how much I missed, but it should be pretty fine now for QGIS 4 for the TWW plugin. Qt6 build is also tested in the plugin CI.
